src/posthog-core.js

```diff
@@ -916,7 +916,7 @@ PostHogLib.prototype.reset = function (reset_device_id) {
     )
 
     // trigger onFeatureFlags callback
-    this.persistence.receivedFeatureFlags()
+    this.featureFlags.reloadFeatureFlags()
```

Comment:
Maybe even just `this.reloadFeatureFlags()`? Not sure why it wasn't called before